contentcuration/contentcuration/views/base.py

```diff
@@ -204,16 +211,37 @@ def get_user_edit_channels(request):
     edit_channels = request.user.editable_channels.exclude(deleted=True)\
         .select_related('main_tree').prefetch_related('editors')\
         .defer('trash_tree', 'clipboard_tree', 'staging_tree', 'chef_tree', 'previous_tree', 'viewers')
+    edit_channels = _apply_channel_filters(edit_channels, request.query_params)
     channel_serializer = AltChannelListSerializer(edit_channels, many=True)
     return Response(channel_serializer.data)
 
 
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAuthenticated,))
+def get_user_channel_sets(request):
+    sets = request.user.channel_sets.prefetch_related('secret_token__channels', 'editors').select_related('secret_token')
+    channelset_serializer = ChannelSetSerializer(sets, many=True)
+    return Response(channelset_serializer.data)
+
+
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAuthenticated,))
+def get_channels_by_token(request, token):
+    channels = Channel.objects.filter(secret_tokens__token=token, deleted=False)
+    channels = _apply_channel_filters(channels, request.query_params)
+    channel_serializer = AltChannelListSerializer(channels, many=True)
```

Comment:
This `AltChannelListSerializer` fetches the resource counts, making another roundtrip query to the DB.

1. is that count necessary for that endpoint? If it's not, can we defer that resource count attribute somehow?
2. If it is necessary, can we have it preloaded using `prefetch` or `select_related`?

Reply:
I'll check if this is even needed in the first place- can't remember now